cli/azd/pkg/tools/docker/container_image.go

```diff
@@ -0,0 +1,86 @@
+package docker
+
+import (
+	"errors"
+	"strings"
+)
+
+// ContainerImage represents a container image and its components
+type ContainerImage struct {
+	// The registry name
+	Registry string
+	// The repository name or path
+	Repository string
+	// The tag
+	Tag string
+}
+
+// Local returns the local image name without registry
+func (ci *ContainerImage) Local() string {
+	builder := strings.Builder{}
+
+	if ci.Repository != "" {
+		builder.WriteString(ci.Repository)
+	}
+
+	if ci.Tag != "" {
+		builder.WriteString(":")
+		builder.WriteString(ci.Tag)
+	}
+
+	return builder.String()
+}
+
+// Remote returns the remote image name with registry when specified
+func (ci *ContainerImage) Remote() string {
+	builder := strings.Builder{}
+	if ci.Registry != "" {
+		builder.WriteString(ci.Registry)
+		builder.WriteString("/")
+	}
+
+	if ci.Repository != "" {
+		builder.WriteString(ci.Repository)
+	}
+
+	if ci.Tag != "" {
+		builder.WriteString(":")
+		builder.WriteString(ci.Tag)
+	}
+
+	return builder.String()
+}
+
+func ParseContainerImage(image string) (*ContainerImage, error) {
+	// Check if the imageURL is empty
+	if image == "" {
+		return nil, errors.New("empty image URL provided")
+	}
+
+	containerImage := &ContainerImage{}
+
+	// Detect tags
+	tagParts := strings.Split(image, ":")
+	if len(tagParts) > 2 {
+		return containerImage, errors.New("invalid tag format")
+	}
```

Comment:
This may fail to parse `myregistryhost:5000/fedora/httpd:version1.0`. See [docker tags](https://docs.docker.com/engine/reference/commandline/image_tag/#description). I implemented something similar to `docker.SplitDockerImage`.